common/reconciliation/invariants/concreteExecutionExists.go

```diff
@@ -115,3 +142,44 @@ func (c *concreteExecutionExists) Fix(execution interface{}) common.FixResult {
 func (c *concreteExecutionExists) InvariantType() common.InvariantType {
 	return common.ConcreteExecutionExistsInvariantType
 }
+
+func (c *concreteExecutionExists) validateCurrentRunID(
+	currentExecution *common.CurrentExecution,
+) (*common.CurrentExecution, *common.CheckResult) {
+
+	resp, err := c.pr.GetCurrentExecution(&persistence.GetCurrentExecutionRequest{
+		DomainID:   currentExecution.DomainID,
+		WorkflowID: currentExecution.WorkflowID,
+	})
+	if err != nil {
+		switch err.(type) {
+		case *shared.EntityNotExistsError:
+			return nil, &common.CheckResult{
+				CheckResultType: common.CheckResultTypeHealthy,
+				InvariantType:   c.InvariantType(),
+				Info:            "current execution does not exist.",
+				InfoDetails:     err.Error(),
+			}
+		default:
+			return nil, &common.CheckResult{
+				CheckResultType: common.CheckResultTypeFailed,
+				InvariantType:   c.InvariantType(),
+				Info:            "failed to get current execution.",
+				InfoDetails:     err.Error(),
+			}
+		}
+	}
+
+	if len(currentExecution.CurrentRunID) == 0 {
+		currentExecution.CurrentRunID = resp.RunID
+	}
+
+	if currentExecution.CurrentRunID != resp.RunID {
+		return nil, &common.CheckResult{
+			CheckResultType: common.CheckResultTypeFailed,
```

Comment:
actual I think we can return healthy here? Seems like it's the same thing as current record with this runID doesn't exist.

Reply:
Yes. Updated